coderd/database/dbauthz/dbauthz_test.go

```diff
@@ -5590,7 +5590,17 @@ func (s *MethodTestSuite) TestAuthorizePrebuiltWorkspace() {
 			Reason:            database.BuildReasonInitiator,
 			TemplateVersionID: tv.ID,
 			JobID:             pj.ID,
-		}).Asserts(w.AsPrebuild(), policy.ActionDelete)
+		}).
+			// Simulate a fallback authorization flow:
+			// - First, the default workspace authorization fails (simulated by returning an error).
+			// - Then, authorization is retried using the prebuilt workspace object, which succeeds.
+			// The test asserts that both authorization attempts occur in the correct order.
+			WithSuccessAuthorizer(func(ctx context.Context, subject rbac.Subject, action policy.Action, obj rbac.Object) error {
+				if obj.Type == rbac.ResourceWorkspace.Type {
+					return xerrors.Errorf("not authorized for workspace type")
+				}
+				return nil
+			}).Asserts(w, policy.ActionDelete, w.AsPrebuild(), policy.ActionDelete)
```

Comment:
Do we also have tests for the converse case?

Reply:
The converse case would be when the workspace authorization check succeeds and the prebuilt check is never called. 
This test is meant to cover the prebuilt authorization check (workspace check fails and it falls back to the prebuilt authorization check). There is already a test that covers just the workspace delete authorization check: https://github.com/coder/coder/blob/7254c08af49c1f7e706ae5343d659f2d3b39067e/coderd/database/dbauthz/dbauthz_test.go#L2863-L2890